enigmatry-entry-blueprint-app/theming-setup.md

```diff
@@ -0,0 +1,233 @@
+# Theming Setup
+
+- [Overview](#overview)
+- [Required packages](#required-packages)
+- [Include styles generator](#include-styles-generator)
+- [Theme](#theme)
+  - [Define custom theme](#define-custom-theme)
+  - [Theme sections and properties](#theme-sections-and-properties)
+  - [Fonts](#fonts)
+  - [Set Angular Material typography](#set-angular-material-typography)
+- [Default values table](#default-values-table)
+
+## Overview
+
+This guide explains how to set up your project to begin using theming. The Entry theme works hand-in-hand with [Angular Material](https://material.angular.io/). It allows you to configure only certain things for simplicity and quick setup.
+
+### Prerequisites:
+1. Install required [`@enigmatry packages`](#required-packages)
+2. Including [`generator`](#include-styles-generator) for theming setup
+3. Prepare custom fonts, more info [`here`](#2-preparing-custom-fonts)
+
+## Required packages
+
+Ensure that the latest version of the[`@enigmatry packages`](#enigmatry-packages) are installed:
+
+- [`entry-components`](https://github.com/enigmatry/entry-angular-building-blocks/tree/master/libs/entry-components)
+- [`entry-form`](https://github.com/enigmatry/entry-angular-building-blocks/tree/master/libs/entry-form)
+
+### @enigmatry/entry-components
+Reusable Entry components for Angular material. It comes with the default Roboto font. There are plenty of components that this library includes: button, dialog, file-input, search-filter, spinner, table, and many more. <br>
+Visit components demo app to check what is currently supported [here](https://entry-demo.enigmatry.com/).
+
+#### Installation
+
+```ts
+npm install @enigmatry/entry-components
+```
+
+### @enigmatry/entry-form
+Form components and helper methods used by entry-code-generation.
+
+#### Installation
+
+```ts
+npm install @enigmatry/entry-form
+```
+
+Import the `EntryFormModule` in your `feature.module` or `shared.module`
+
+```typescript
+import { EntryFormModule } from '@enigmatry/entry-form';
+```
+
+## Include styles generator
+
+[`Entry components library`](#required-packages) comes with a generator file. The generator includes multiple generator files for supported components (dialog, table, button, form...)
+Call it in your dedicated SCSS file for theming.
+
+```scss
+@use 'entry-components/styles/generator';
```

Comment:
For this to work, it is expected to add shortcut in angular.json towards node_modules/@enigmatry. This has to be mentioned.